share/functions/edit_command_buffer.fish

```diff
@@ -0,0 +1,45 @@
+function edit_command_buffer --description 'Edit the command buffer in an external editor'
+    set -l f (mktemp)
+    if set -q f[1]
+        mv $f $f.fish
+        set f $f.fish
+    else
+        # We should never execute this block but better to be paranoid.
+        if set -q TMPDIR
+            set f $TMPDIR/fish.(echo %self).fish
+        else
+            set f /tmp/fish.(echo %self).fish
+        end
+        touch $f
+        or return 1
+    end
+
+    # Edit the command line with the users preferred editor or vim or emacs.
+    commandline -b >$f
+    if set -q EDITOR
+        eval $EDITOR $f
+    else if command -s vim >/dev/null
+        vim $f
+    else if command -s emacs >/dev/null
+        emacs $f
+    else
+        echo
+        echo (_ 'External editor requested but $EDITOR not set, and vim or emacs not available.')
+        echo (_ 'Try setting the EDITOR variable to a text editor, or installing vim or emacs.')
+        commandline -f repaint
+        command rm $f
+        return 1
+    end
+
+    if test $status -eq 0 -a -s $f
+        # Set the command to the output of the edited command and move the cursor to the
+        # end of the edited command.
+        commandline -r (cat $f)
+        commandline -C 999999
```

Comment:
@flaom, Perhaps you should open a new issue, and take ownership of it, rather than commenting on a closed PR where your proposal has zero chance of being implemented?